### PR TITLE
Fix deprecated CURL API usage.

### PR DESCRIPTION
**Pull request by utoni**
```
lib/./http/HttpClient_Curl.hpp: In member function ‘long int Microsoft::Applications::Events::CurlHttpOperation::Send()’:
lib/./http/HttpClient_Curl.hpp:211:39: error: ‘CURLINFO_LASTSOCKET’ is deprecated: since 7.45.0. Use CURLINFO_ACTIVESOCKET [-Werror=deprecated-declarations]
  211 |         res = curl_easy_getinfo(curl, CURLINFO_LASTSOCKET, &sockextr);
```

### CODE REVIEW

File: lib/http/HttpClient_Curl.hpp [hunk 1]
```diff
@@ -208,7 +208,7 @@ class CurlHttpOperation {
          * Note that this API takes a pointer to a 'long' while we use
          * curl_socket_t for sockets otherwise.
          */
-        res = curl_easy_getinfo(curl, CURLINFO_LASTSOCKET, &sockextr);
+        res = curl_easy_getinfo(curl, CURLINFO_ACTIVESOCKET, &sockextr);
```
Comment (lalitb) [inline lib/http/HttpClient_Curl.hpp:211]: CURLINFO_ACTIVESOCKET was added in  7.45.0. Just to be on safer side, should we check for the libcurl version (probably using [LIBCURL_VERSION_NUM](https://github.com/curl/curl/blob/aee4ebe59161d0a5281743f96e7738ad97fe1cd4/include/curl/curlver.h#L62)), and use right one accordingly.

**Review by lalitb — APPROVED** [review]
Approve with suggestion :)

File: lib/http/HttpClient_Curl.hpp (hunk 1 above)
Comment (utoni) [reply lib/http/HttpClient_Curl.hpp:211]: Should I change that in the current PR or a follow-up? :)

Comment (ThomsonTan) [reply lib/http/HttpClient_Curl.hpp:211]: Add a TODO comment here if it will be included in the following up PR?

**Comment by utoni** [comment]
Fixed for the sake of simplicity.

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**